src/plugins/mer/merhardwaredevicewizardpages.cpp

```diff
@@ -138,34 +139,40 @@ void MerHardwareDeviceWizardSelectionPage::handleTestConnectionClicked()
     sshParams.timeout = timeout();
     sshParams.authenticationType = SshConnectionParameters::AuthenticationTypeAll;
 
+    QString error;
     m_ui->connectionTestLabel->setText(tr("Connecting to machine %1 ...").arg(hostName()));
     m_ui->connectionTestLabel->setText(MerConnectionManager::testConnection(sshParams,
                 &m_connectionTestOk));
     if (!m_connectionTestOk)
         goto end;
 
     m_ui->connectionTestLabel->setText(tr("Detecting device properties..."));
-    m_architecture = detectArchitecture(sshParams, &m_connectionTestOk);
-    m_deviceName = detectDeviceName(sshParams, &m_connectionTestOk);
+    m_architecture = detectArchitecture(sshParams, &m_connectionTestOk, &error);
+    if (m_connectionTestOk) {
+        m_deviceName = detectDeviceName(sshParams, &m_connectionTestOk, &error);
+    }
     if (!m_connectionTestOk) {
         m_ui->connectionTestLabel->setText(tr("Could not autodetect device properties"));
+        QMessageBox::critical(this, tr("Device Error"), error);
         goto end;
     }
 
-    m_ui->connectionTestLabel->setText(tr("Detecting SDK tools..."));
-    m_sdkClientToolsInstalled = detectSdkClientTools(sshParams, &m_connectionTestOk);
+    m_ui->connectionTestLabel->setText(tr("Detecting software packages required for proper SDK operation..."));
+    m_sdkClientToolsInstalled = detectSdkClientTools(sshParams, &m_connectionTestOk, &error);
     if (!m_connectionTestOk) {
-        m_ui->connectionTestLabel->setText(tr("Could not autodetect SDK tools"));
+        m_ui->connectionTestLabel->setText(tr("Could not autodetect installed software packages"));
+        QMessageBox::critical(this, tr("Device Error"), error);
         goto end;
     }
     if (!m_sdkClientToolsInstalled) {
-        m_ui->connectionTestLabel->setText(tr("SDK tools are not installed"));
+        m_ui->connectionTestLabel->setText(tr("Software packages required for proper SDK operation missing"));
         QString password;
         if (askInstallSdkClientTools(&password)) {
-            m_ui->connectionTestLabel->setText(tr("Installing SDK tools..."));
-            m_sdkClientToolsInstalled = installSdkClientTools(sshParams, password);
+            m_ui->connectionTestLabel->setText(tr("Installing software packages..."));
```

Comment:
I wouldn't worry about repeating the full "software packages required for proper SDK operation" phrase in every message. There is no context permanently available that could help the user guess the missing part from.

Reply:
I think in this case user must have just approved installation of those packages so they must be aware that it's "software packages required for proper SDK operation" they were installing. That's why I decided to leave it out and it's a bit too long to fit on one line so that's why I tried to shorten it.

Reply:
True :) Please check if the other cases would profit from full wording (I didn't)

Reply:
I think it's now quite clear for the user. Error messages and the install approval dialog contain the full wording.